Core/include/Acts/Propagator/PropagatorOptions.hpp

```diff
@@ -30,6 +30,9 @@ struct PurePropagatorPlainOptions {
   /// Maximum number of steps for one propagate call
   unsigned int maxSteps = 1000;
 
+  /// Maximum number of next target calls for one step
```

Comment:
Maybe we could elaborate more what this protects against, and why this might happen.